### DISCUSSION TITLE
OpenThread router on Linux PC

### DISCUSSION THREAD

**Discussion by sohaibbinaltaf** [openthread/openthread#8628] [Q&A]
I have developed an openthread network using Raspberry Pis (Raspbian) and nRF52 dongle from Nordic. Now I am trying to make a similar router by installing the openthread border router settings on my PC, on VMware with Ubuntu 22.04.1 LTS installed over it. The dongle has ot-rcp.hex file burned.

I was installing the repositories and when i reached the step to verify, sudo systemctl status command gives State: running. But 

sudo service otbr-agent status. returns 

otbr-agent.service - OpenThread Border Router Agent
     Loaded: loaded (/lib/systemd/system/otbr-agent.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-01-08 16:47:05 +03; 3s ago
    Process: 5007 ExecStartPre=service mdns start (code=exited, status=0/SUCCESS)
    Process: 5013 ExecStart=/usr/sbin/otbr-agent $OTBR_AGENT_OPTS **(code=exited, status=1/FAILURE)**
   Main PID: 5013 (code=exited, status=1/FAILURE)
        CPU: 31ms

is there something missing which is required to be installed?

**Comment by jwhui** [comment]
Can you provide the `otbr-agent` log output?
```
cat /var/log/syslog | grep otbr-agent
```

**Reply by sohaibbinaltaf** [reply]
sorry I was away from my project due to some reasons. let me check again, I will share the log output here

**Reply by sohaibbinaltaf** [reply]
here is the log output: 

[log file.txt](https://github.com/openthread/openthread/files/10695010/log.file.txt)

**Reply by jwhui** [reply]
Can you verify whether `/dev/ttyACM0` appears when the USB dongle is plugged in?

**Reply by sohaibbinaltaf** [reply]
using ls /dev/tty* I was able to view a list and it also had this /dev/ttyACM0

**Reply by jwhui** [reply]
The following error:
```
Feb  9 10:57:24 sohaib-virtual-machine otbr-agent[1341]: 49d.17:02:54.132 [C] Platform------: Init() at hdlc_interface.cpp:151: InvalidArgument
```

indicates that calling `stat` with that path fails.

https://github.com/openthread/openthread/blob/9ff2eefee0ce59d8a006e9d32761cc25a9015e63/src/posix/platform/hdlc_interface.cpp#L151

**Reply by sohaibbinaltaf** [reply]
How can this issue be solved? Is something missing or required to be installed

**Reply by jwhui** [reply]
If `stat` is failing, there is an issue with the device path. Have you tried looking at `errno`?

**Comment by joda58** [comment]
Hi! 
I started out with the Guides... and got the same error message.

BTW: I am running on Ubuntu 22.10 (x64).

First time during the guide: **_"Simulating a Thread network with OpenThread"_**
    I can not run **Step 6, Item 2:**  _sudo ./build/posix/src/posix/ot-daemon -v \
                                'spinel+hdlc+forkpty:///build/simulation/examples/apps/ncp/ot-rcp?forkpty-arg=2'_

> ./build/posix/src/posix/ot-daemon[176425]: Running OPENTHREAD/afbb2d579; POSIX; Feb 18 2023 15:13:11
> ./build/posix/src/posix/ot-daemon[176425]: Thread version: 4
> ./build/posix/src/posix/ot-daemon[176425]: **49d.19:42:26.578 [C] Platform------: Init() at hdlc_interface.cpp:151: InvalidArgument**


Second time during the guide: **_"Thread Border Router - Bidirectional IPv6 Connectivity and DNS-Based Service Discovery"_**
    I can not run **Step 2 , Check if OTBR is successfully installed :**  _sudo service otbr-agent status_

○ otbr-agent.service - OpenThread Border Router Agent
     Loaded: loaded (/lib/systemd/system/otbr-agent.service; enabled; preset: enabled)
     Active: inactive (dead)

> feb 18 16:58:16 UbuntuMate systemd[1]: otbr-agent.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [NOTE]-AGENT---: Thread version: 1.3.0
> feb 18 16:58:16 UbuntuMate systemd[1]: otbr-agent.service: Failed with result 'exit-code'.
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [NOTE]-AGENT---: Thread interface: wpan0
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [NOTE]-AGENT---: Radio URL: spinel+hdlc+uart:///dev/ttyACM0
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [NOTE]-AGENT---: Radio URL: trel://enp8s0
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [NOTE]-ILS-----: Infra link selected: enp8s0
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: [INFO]-NCP-----: OpenThread log level changed to 4
> feb 18 16:58:16 UbuntuMate otbr-agent[255938]: **_49d.20:51:07.082 [C] Platform------: Init() at hdlc_interface.cpp:151: InvalidArgument_**
> feb 18 16:58:18 UbuntuMate systemd[1]: Stopped OpenThread Border Router Agent.

Please help! 

/JD

**Reply by Irving-cl** [reply]
Hi @joda58,

For the first issue, could you check if you have built the simulation RCP? It should be at `build/simulation/examples/apps/ncp/ot-rcp`. I just have a try, if I removed the simualation RCP, I will get the same error as you got. If I build it again, I can get ot-daemon run:
```
➜  openthread git:(unit_test_readme) sudo ./build/posix/src/posix/ot-daemon -v 'spinel+hdlc+forkpty://build/simulation/examples/apps/ncp/ot-rcp?forkpty-arg=2'
./build/posix/src/posix/ot-daemon[2101894]: Running OPENTHREAD/thread-reference-20200818-2326-gb8e1a37b4; POSIX; Feb 24 2023 11:09:01
./build/posix/src/posix/ot-daemon[2101894]: Thread version: 4
./build/posix/src/posix/ot-daemon[2101894]: Thread interface: wpan0
./build/posix/src/posix/ot-daemon[2101894]: RCP version: OPENTHREAD/thread-reference-20200818-2326-gb8e1a37b4; SIMULATION; Feb 24 2023 11:08:06
```

To build the simulation RCP, run:
```
./script/cmake-build simulation -DOT_APP_CLI=OFF -DOT_APP_NCP=OFF -DOT_FTD=OFF -DOT_MTD=OF
```

This page may also be helpful: https://github.com/openthread/openthread/blob/main/src/posix/README.md

---

For the second issue, since the URL is `/dev/ttyACM0`, you are using a real RCP. Could you check if the RCP device is correctly connected on your PC? Are you using a nRF52840 DK?

**Reply by joda58** [reply]
Hi Irving!

Thanks for your help .....

The first issue was problems with an extra '/' on the OpenThread web page.
In your email you had two ('spinel+hdlc+forkpty://build...) but web page had three.
:-)

The second issue I think was solve by reading up on things ... and with the help 
of your info I have learned some new things.

Many Thanks!

/JD
